test/e2e/federated-service.go

```diff
@@ -47,239 +49,315 @@ const (
 
 	FederatedServiceTimeout = 60 * time.Second
 
-	FederatedServiceName = "federated-service"
-	FederatedServicePod  = "federated-service-test-pod"
+	FederatedServiceName    = "federated-service"
+	FederatedServicePodName = "federated-service-test-pod"
 
 	DefaultFederationName = "federation"
 
 	// We use this to decide how long to wait for our DNS probes to succeed.
-	DNSTTL = 180 * time.Second
+	DNSTTL = 180 * time.Second // TODO: make k8s.io/kubernetes/federation/pkg/federation-controller/service.minDnsTtl exported, and import it here.
 )
 
 var FederatedServiceLabels = map[string]string{
 	"foo": "bar",
 }
 
-var _ = framework.KubeDescribe("[Feature:Federation] Federated Services", func() {
-
+var _ = framework.KubeDescribe("[Feature:Federation]", func() {
+	f := framework.NewDefaultFederatedFramework("federated-service")
 	var clusterClientSets []*release_1_3.Clientset
+	var clusterNamespaceCreated []bool // Did we need to create a new namespace in each of the above clusters?  If so, we should delete it.
```

Comment:
FYI, I'll be converting all this to a single map keyed on cluster name, and this list of bools will become a field in the struct.